### ISSUE TITLE
Single token generation V3.0 bug [V2.0 -> v3.0]

### ISSUE THREAD

**Issue by Lurrobert** [kevinhermawan/OllamaKit#7]
Hey! so I have accidentally installed V3.0 and my token generation got broken

I was generating a single requisition with this code

Hope it will save someone time

```
if await ollamaKit.reachable() {
            let data = message.convertToOKGenerateRequestData()
            
            generation = ollamaKit.generate(data: data)
                .sink(receiveCompletion: { [weak self] completion in
                    switch completion {
                    case .finished:
                        self?.handleComplete()
                    case .failure(let error):
                        self?.handleError(error.localizedDescription)
                    }
                }, receiveValue: { [weak self] response in
                    self?.handleReceive(response)
                })
        } else {
            self.handleError(AppMessages.ollamaServerUnreachable)
        }
```

So this piece of code does not work any more in 3.0 in generate one token

![CleanShot 2024-01-15 at 08 44 15@2x](https://github.com/kevinhermawan/OllamaKit/assets/40468118/4c3b5ac8-0461-4fbc-bbec-ed4a1531b42e)

**Comment by Lurrobert** [comment]
I expect this to be Future token at this PR

https://github.com/kevinhermawan/OllamaKit/commit/b4d1fc5bdb34a07d23e1a9a3ef6548287dada15a

![CleanShot 2024-01-15 at 09 23 01@2x](https://github.com/kevinhermawan/OllamaKit/assets/40468118/686255b7-f1e1-4e7b-9bad-89467829437e)

**Comment by kevinhermawan** [comment]
Hi @Lurrobert, I've refactored the code from `Future` to `PassthroughSubject`. Can you please try it from the main branch and let me know if it resolves your issue?

**Comment by Lurrobert** [comment]
@kevinhermawan 
not sure if this is by design or intent but this also appeared, could easily quick fix but just letting you know
![CleanShot 2024-01-15 at 11 47 38@2x](https://github.com/kevinhermawan/OllamaKit/assets/40468118/79d787f8-c04c-43bf-aa4a-d6ae8fa09279)

**Comment by Lurrobert** [comment]
perfect, confirm its working!

**Comment by kevinhermawan** [comment]
> @kevinhermawan not sure if this is by design or intent but this also appeared, could easily quick fix but just letting you know ![CleanShot 2024-01-15 at 11 47 38@2x](https://private-user-images.githubusercontent.com/40468118/296732281-79d787f8-c04c-43bf-aa4a-d6ae8fa09279.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUzMTYzMjUsIm5iZiI6MTcwNTMxNjAyNSwicGF0aCI6Ii80MDQ2ODExOC8yOTY3MzIyODEtNzlkNzg3ZjgtYzA0Yy00M2JmLWFhNGEtZDZhZThmYTA5Mjc5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE1VDEwNTM0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVkZTg5N2IzZTc3YzIxMTY2YjI4M2M5NjdmNzNiMWJmMzcwYWI1MDRhMjE0ZGFmYTAyYTY3NzM0ZGE4OWQ2MTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DjWSFgne8FvohktJRiKwycU39VfORnQ7cb25iPGVr3U)
> 
> qf: `data.context = self.context?.map { Double($0) }`

Ah, thanks, mate. If I see the Ollama code, it needs to be an `integer` and not a `double`. I will create a new PR to fix that.

https://github.com/jmorganca/ollama/blob/8795447dad8f056a695eafead898db4608c98014/api/types.go#L41C12-L41C12

**Comment by kevinhermawan** [comment]
Fixed in [version 3.0.1](https://github.com/kevinhermawan/OllamaKit/releases/tag/3.0.1)
